Fix select icons

aptkingston opened the pull request:
## Description
This is a quick fix for some issues with the new icon functionality in selects. I had raised these issues in the OIDC PR but I was off for a few days and didn't see the merge, so some wee things must have just got lost in the review comments.

In short - the `getOptionIcon` default prop needs to be a function as it is invoked without checking for being nullish. Currently the default is just `null` in the core Select and Picker components, which means any direct usage of these will throw a JS error as soon as a value is picked. This is most evident from the client apps, where it's no longer possible to use an `options` or `relationship` field. All I've done is change the default prop to be what I had suggested in the review, `() => null`.

The other quick fix is that because icons are shown by default for any object which have an `icon` property, icons were trying to appear when creating a new column, as shown here:
![image](https://user-images.githubusercontent.com/9075550/126499432-6c5f98e1-86cc-4f0a-a7b6-f782f93594b0.png)

This is because the definition of these options include `icon` properties for some reason (and old remix icons, at that), which is likely just a leftover remnant from a previous feature. I've just removed the `icon` property from the definition to prevent the select from attempting to render invalid icons.

I've also updated the `alt` description of the icons to be shorter and more generic, but hopefully this never gets displayed again anyway.

PClmnt commented:
Hey, sorry about this, that's my bad. I thought i addressed the PR comments but missed these.

aptkingston commented:
> 
> 
> Hey, sorry about this, that's my bad. I thought i addressed the PR comments but missed these.

All good don't worry! It was a huge PR, easy to miss a couple of tiny things like this.

kevmodrome reviewed (approved):
LGTM
